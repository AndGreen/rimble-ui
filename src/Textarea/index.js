import React from 'react'
import styled from 'styled-components'
import { space } from 'styled-system'

import Input from '../Input'

const Textarea = styled(Input)`
  appearance: none;
  font-size: 1rem;
  line-height: 1;
  height: auto;
  background: #FFFFFF;

  ${space}
`

Textarea.defaultProps = {
  as: 'textarea',
  p: 3,
  borders: 1,
  borderColor: 'grey',
  borderRadius: 1,
  boxShadow: 1
}

Textarea.displayName = 'Textarea'

export default Textarea
