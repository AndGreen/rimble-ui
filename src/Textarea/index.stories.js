import React from 'react'

import { storiesOf } from '@storybook/react'

import Textarea from '../Textarea'

storiesOf('Form/Textarea', module)
  .add('default', () => (
    <Textarea rows={4} defaultValue='Hello' />
  ))
