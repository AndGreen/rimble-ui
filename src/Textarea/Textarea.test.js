import Textarea from './'

describe('Textarea component sanity', () => {
  it('has name', () => {
      expect(Textarea.displayName).toBe('Textarea')
  })

  it('matches default snapshot', () => {
    const component = render(<Textarea />)
    expect(component).toMatchSnapshot()
  })
})
