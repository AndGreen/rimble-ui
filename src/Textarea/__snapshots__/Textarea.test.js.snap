// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea component sanity matches default snapshot 1`] = `
.c0 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: 1rem;
  line-height: 1;
  height: 48px;
  padding: 0 1rem;
  border: 1px solid;
  background: #FFFFFF;
  border-color: #CCC;
  border-radius: 4px;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.1);
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: 1rem;
  line-height: 1;
  height: auto;
  background: #FFFFFF;
  padding: 16px;
}

.c0:hover {
  box-shadow: 0px 2px 6px rgba(0,0,0,0.3);
}

.c0:focus {
  outline: none;
  border-color: #5436D6;
}

.c0:disabled {
  opacity: 0.4;
  pointer-events: none;
}

<textarea
  class="c0"
>
  
</textarea>
`;
